src/sentry/utils/arroyo.py

```diff
@@ -51,3 +62,103 @@ def timing(self, name: str, value: Union[int, float], tags: Optional[Tags] = Non
+    initializer: Optional[Callable[[], None]]
+) -> Callable[[], None]:
+    from sentry.metrics.middleware import get_current_global_tags
+
+    # One can add integer tags and other invalid types today. Filter out any
+    # tags that may not be pickleable. Because those tags are getting pickled
+    # as part of the constructed partial()
+    tags: Tags = {k: v for k, v in get_current_global_tags().items() if isinstance(v, str)}
+    return partial(_initialize_arroyo_subprocess, initializer=initializer, tags=tags)
+
+
+def _initialize_arroyo_subprocess(initializer: Optional[Callable[[], None]], tags: Tags) -> None:
+    from sentry.runner import configure
+
+    configure()
+
+    if initializer:
+        initializer()
+
+    from sentry.metrics.middleware import add_global_tags
+
+    # Inherit global tags from the parent process
+    add_global_tags(_all_threads=True, **tags)
+
+
+def _initialize_arroyo_main() -> None:
+    from arroyo import configure_metrics
+
+    from sentry.utils.metrics import backend
+
+    metrics_wrapper = MetricsWrapper(backend, name="consumer")
+    configure_metrics(metrics_wrapper)
+
+
+class RunTaskWithMultiprocessing(ProcessingStrategy[Union[FilteredPayload, TStrategyPayload]]):
+    def __new__(
+        cls,
+        *function: Callable[[Message[TStrategyPayload]], TResult],
+        next_step: ProcessingStrategy[Union[FilteredPayload, TResult]],
+        initializer: Optional[Callable[[], None]] = None,
+        **kwargs: Any,
+    ) -> RunTaskWithMultiprocessing[Union[FilteredPayload, TStrategyPayload]]:
+
+        from django.conf import settings
+
+        if settings.KAFKA_CONSUMER_FORCE_DISABLE_MULTIPROCESSING:
+            from arroyo.processing.strategies.run_task import RunTask
+
+            return RunTask(**kwargs)  # type: ignore[return-value]
+        else:
+            from arroyo.processing.strategies.run_task_with_multiprocessing import (
+                RunTaskWithMultiprocessing as ArroyoRunTaskWithMultiprocessing,
+            )
+
+            return ArroyoRunTaskWithMultiprocessing(  # type: ignore[return-value]
+                initializer=_get_arroyo_subprocess_initializer(initializer), **kwargs
+            )
+
+
+def run_basic_consumer(
+    topic: str,
+    group_id: str,
+    auto_offset_reset: str,
+    strict_offset_reset: bool,
+    force_topic: str | None,
+    force_cluster: str | None,
+    strategy_factory_cls: Type[ProcessingStrategyFactory[Any]],
+) -> None:
+    from django.conf import settings
+
+    from sentry.metrics.middleware import add_global_tags
+
+    add_global_tags(kafka_topic=topic, group_id=group_id)
+
+    topic = force_topic or topic
+
+    cluster_name: str = force_cluster or settings.KAFKA_TOPICS[topic]["cluster"]
+    consumer_config = build_kafka_consumer_configuration(
+        kafka_config.get_kafka_consumer_cluster_options(
+            cluster_name,
+        ),
+        group_id=group_id,
+        auto_offset_reset="latest",
```

Comment:
is this to be swapped out? i guess we should also apply the strict offset reset setting passed in?